ebos/ecloutputblackoilmodule.hh

```diff
@@ -579,14 +587,75 @@ public:
                         val.second = getValue(fs.viscosity(gasPhaseIdx));
                     else if (key.first == "BVOIL" || key.first == "BOVIS")
                         val.second = getValue(fs.viscosity(oilPhaseIdx));
-                    else if (key.first == "BRPV")
-                        val.second = elemCtx.simulator().model().dofTotalVolume(globalDofIdx)*getValue(intQuants.porosity());
-                    else if (key.first == "BOPV")
-                        val.second = getValue(fs.saturation(oilPhaseIdx))*elemCtx.simulator().model().dofTotalVolume(globalDofIdx)*getValue(intQuants.porosity());
-                    else if (key.first == "BWPV")
-                        val.second = getValue(fs.saturation(waterPhaseIdx))*elemCtx.simulator().model().dofTotalVolume(globalDofIdx)*getValue(intQuants.porosity());
-                    else if (key.first == "BGPV")
-                        val.second = getValue(fs.saturation(gasPhaseIdx))*elemCtx.simulator().model().dofTotalVolume(globalDofIdx)*getValue(intQuants.porosity());
+                    else if ((key.first == "BRPV") ||
+                             (key.first == "BOPV") ||
+                             (key.first == "BWPV") ||
+                             (key.first == "BGPV"))
+                    {
+                        if (key.first == "BRPV") {
+                            val.second = 1.0;
+                        }
+                        else if (key.first == "BOPV") {
+                            val.second = getValue(fs.saturation(oilPhaseIdx));
+                        }
+                        else if (key.first == "BWPV") {
+                            val.second = getValue(fs.saturation(waterPhaseIdx));
+                        }
+                        else {
+                            val.second = getValue(fs.saturation(gasPhaseIdx));
+                        }
+
+                        // Include active pore-volume.
+                        val.second *= elemCtx.simulator().model().dofTotalVolume(globalDofIdx)
+                            * getValue(intQuants.porosity());
+                    }
```

Comment:
Might be personal taste, but I don't see an improvement here. Just more branches. feel free to ignore, though

Reply:
> I don't see an improvement here, just more branches.

That's fair, I guess.  I still consider it an improvement because I don't have to repeat the active pore-volume expression for all cases.  Doing it this way is less maintenance in my opinion.

Reply:
Well, we disagree. One could also save some lines:
```
al.second = elemCtx.simulator().model().dofTotalVolume(globalDofIdx)
                            * getValue(intQuants.porosity());
if (key.first == "BOPV") {
    val.second *= getValue(fs.saturation(oilPhaseIdx));
}
else if (key.first == "BWPV") {
      val.second *= getValue(fs.saturation(waterPhaseIdx));
}
else {
      val.second *= getValue(fs.saturation(gasPhaseIdx));
}
```

Reply:
and now that i look at it I see that the above statement is wrong.